lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -2728,7 +2728,7 @@ ThreadContext::InvalidateInlineCacheList(InlineCacheList* inlineCacheList)
         }
     }
     NEXT_SLISTBASE_ENTRY;
-    inlineCacheList->Clear();
+    Adelete(&this->inlineCacheThreadInfoAllocator, inlineCacheList);
```

Comment:
The list is a list of pointer to inlineCache, deleting the list doesn't delete the inlineCache.
Zeroing the inlineCache will clear the now invalid information on it so tit can be used again.